packages/blocks/src/root-block/edgeless/edgeless-root-block.ts

```diff
@@ -149,10 +150,22 @@ export class EdgelessRootBlockComponent extends BlockElement<
 
   mouseRoot!: HTMLElement;
 
-  @state()
-  accessor edgelessTool: EdgelessTool = {
-    type: localStorage.defaultTool ?? 'default',
-  };
+  private _edgelessToolProvider = new ContextProvider(this, {
```

Comment:
I still don't think adding the provider to the EdgelessRootBlock is necessary. Maybe we could narrow its usage to the toolbar

Reply:
Okay, have removed from edgeless-root-block